python/pyspark/sql/session.py

```diff
@@ -394,6 +394,36 @@ def enableHiveSupport(self) -> "SparkSession.Builder":
             """
             return self.config("spark.sql.catalogImplementation", "hive")
 
+        def create(self) -> "SparkSession":
+            """Creates a new SparkSession. Can only be used in the context of Spark Connect
+            and will throw an exception otherwise.
+
+            .. versionadded:: 3.5.0
+
+            Returns
+            -------
+            :class:`SparkSession`
+            """
+            opts = dict(self._options)
+            if "SPARK_REMOTE" in os.environ or "spark.remote" in opts:
+                from pyspark.sql.connect.session import SparkSession as RemoteSparkSession
+
+                url = opts.get("spark.remote", os.environ.get("SPARK_REMOTE"))
+
+                if url.startswith("local"):
+                    raise RuntimeError(
+                        "Creating new SparkSessions with `local` "
+                        "connection string is not supported."
+                    )
+
+                # Mark this Spark Session as Spark Connect. This prevents that local PySpark is
+                # used in conjunction with Spark Connect mode.
+                os.environ["SPARK_ENABLE_CONNECT_MODE"] = "1"
+                opts["spark.remote"] = url
+                return RemoteSparkSession.builder.config(map=opts).create()
+            else:
+                raise RuntimeError("Create a new SparkSession is only supported with SparkConnect.")
```

Comment:
Sounds actually a bit awkward.. could we, for example, say "SparkSession.builder.create() is only for Spark Connect; however, `spark.remote` address was not found"

We should probably also check if `spark.master` is specified too, and throw an exception like: "SparkSession.builder.create() is only for Spark Connect; however, `spark.master` address was found"

Reply:
There are some logic to refer at `Sparksession.Builder.config`

Reply:
I reused the logic for validating the URIs and will update the error message.